File: get-started/azure-auth/python/get_user.py
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT License.

# <ProgramSnippet>
import asyncio
from azure.identity import DeviceCodeCredential
from kiota_authentication_azure.azure_identity_authentication_provider import (
    AzureIdentityAuthenticationProvider)
from kiota_http.httpx_request_adapter import HttpxRequestAdapter
from client.get_user_api_client import GetUserApiClient

async def main():
    # You may need this if your're using asyncio on windows
    # See: https://stackoverflow.com/questions/63860576
    # asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())

    client_id = 'YOUR_CLIENT_ID'
    graph_scopes = ['User.Read']

    credential = DeviceCodeCredential(client_id)
    auth_provider = AzureIdentityAuthenticationProvider(credential, scopes=graph_scopes)

    request_adapter = HttpxRequestAdapter(auth_provider)
    client = GetUserApiClient(request_adapter)

    user_me = await client.me.get()
    print(f"Hello {user_me.display_name}, your ID is {user_me.id}")

# Run main
asyncio.run(main())
# </ProgramSnippet>

File: get-started/azure-auth/go/getuser.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

// <ProgramSnippet>
package main

import (
	"context"
	"fmt"

	"getuser/client"

	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
	azure "github.com/microsoft/kiota-authentication-azure-go"
	bundle "github.com/microsoft/kiota-bundle-go"
)

func main() {
	clientId := "YOUR_CLIENT_ID"

	// The auth provider will only authorize requests to
	// the allowed hosts, in this case Microsoft Graph
	allowedHosts := []string{"graph.microsoft.com"}
	graphScopes := []string{"User.Read"}

	credential, err := azidentity.NewDeviceCodeCredential(&azidentity.DeviceCodeCredentialOptions{
		ClientID: clientId,
		UserPrompt: func(ctx context.Context, dcm azidentity.DeviceCodeMessage) error {
			fmt.Println(dcm.Message)
			return nil
		},
	})

	if err != nil {
		fmt.Printf("Error creating credential: %v\n", err)
	}

	authProvider, err := azure.NewAzureIdentityAuthenticationProviderWithScopesAndValidHosts(
		credential, graphScopes, allowedHosts)

	if err != nil {
		fmt.Printf("Error creating auth provider: %v\n", err)
	}

	adapter, err := bundle.NewDefaultRequestAdapter(authProvider)

	if err != nil {
		fmt.Printf("Error creating request adapter: %v\n", err)
	}

	client := client.NewGraphApiClient(adapter)

	me, err := client.Me().Get(context.Background(), nil)

	if err != nil {
		fmt.Printf("Error getting user: %v\n", err)
	}

	fmt.Printf("Hello %s, your ID is %s\n", *me.GetDisplayName(), *me.GetId())
}
// </ProgramSnippet>
